### ISSUE TITLE
RCS1174 conflicts with RCS1229 for methods with `using` declaration

### ISSUE THREAD

**Issue by gtbuchanan** [JosefPihrt/Roslynator#951]
**Product and Version Used**: 4.1.1

**Steps to Reproduce**:
1. Enable [RCS1174] since it's disabled by default
2. `await` a `Task` when returning from a method that also has a `using` declaration.

Note that [RCS1174] is not reported when the `using` declaration is changed to a regular `using` statement.

**Actual Behavior**:
```csharp
private async Task<string> ProcessFile() // RCS1174 & RCS1174FadeOut are reported on `async`
{
    using var stream = File.OpenRead(@"C:\my-file.txt");
    return await this.ReadContents(stream); // RCS1174 & RCS1174FadeOut are reported on `await`
}
```

**Expected Behavior**:
[RCS1174] should not be reported when it would violate [RCS1229]. [RCS1174] should treat the `using` declaration the same as a regular `using` statement.

Relates to #726.

[RCS1174]: https://github.com/JosefPihrt/Roslynator/blob/main/docs/analyzers/RCS1174.md
[RCS1229]: https://github.com/JosefPihrt/Roslynator/blob/main/docs/analyzers/RCS1229.md
